plugins/modules/web_infrastructure/jenkins_plugin.py

```diff
@@ -696,7 +696,8 @@ def _pm_query(self, action, msg):
         self._get_url_data(
             url,
             msg_status="Plugin not found. %s" % url,
-            msg_exception="%s has failed." % msg)
+            msg_exception="%s has failed." % msg,
+            method="POST")
```

Comment:
I'm not sure, but this function is called for `uninstall` and `pinning` as well, how does this change affect them?

Reply:
Hi @aminvakil, thx for your review

Jenkins requires POST requests for enable, disable and uininstall plugins
Pinning plugins feature is not supported in Jenkins 2 https://wiki.jenkins.io/display/JENKINS/Pinned+Plugins

Reply:
> Jenkins requires POST requests for enable, disable and uininstall plugins

Please add `absent` to changelog fragment as well then.
> Pinning plugins feature is not supported in Jenkins 2 https://wiki.jenkins.io/display/JENKINS/Pinned+Plugins

I couldn't find anything in documentation that Jenkins < 2 is not supported, also as `pinned` is still a choice in this module for `state` option, it's better to fix that too IMHO (if it needs fixing).

Reply:
> I couldn't find anything in documentation that Jenkins < 2 is not supported,

https://www.jenkins.io/blog/2016/07/07/jenkins-2/

Reply:
Sorry if I wasn't clear, I meant module documentation.
Even if jenkins 1 is EOL and unsupported, it has not been stated in this module, either change this to support `uninstalled` and `pinned` states or this should be marked as a `breaking_change` and has to wait until 4.0.0.

Reply:
I've added a suggestion for this below.